rust/operator-binary/src/discovery.rs

```diff
@@ -31,180 +26,126 @@ pub enum Error {
+        listener_address
+            .ports
+            .get(HIVE_PORT_NAME)
+            .copied()
+            .context(NoServicePortSnafu {
+                port_name: HIVE_PORT_NAME,
+                role
+            })?
+    );
     if let Some(chroot) = chroot {
         if !chroot.starts_with('/') {
             return RelativeChrootSnafu { chroot }.fail();
         }
         conn_str.push_str(chroot);
     }
-    ConfigMapBuilder::new()
-        .metadata(
-            ObjectMetaBuilder::new()
-                .name_and_namespace(hive)
-                .name(name)
-                .ownerreference_from_resource(owner, None, Some(true))
-                .with_context(|_| ObjectMissingMetadataForOwnerRefSnafu {
-                    hive: ObjectRef::from_obj(hive),
-                })?
-                .with_recommended_labels(build_recommended_labels(
-                    hive,
-                    &resolved_product_image.app_version_label,
-                    &HiveRole::MetaStore.to_string(),
-                    "discovery",
-                ))
-                .context(MetadataBuildSnafu)?
-                .build(),
-        )
-        .add_data("HIVE", conn_str)
-        .build()
-        .with_context(|_| DiscoveryConfigMapSnafu {
-            obj_ref: ObjectRef::from_obj(hive),
-        })
+    Ok(conn_str)
 }
 
-/// Lists all nodes currently hosting Pods participating in the [`Service`]
-async fn nodeport_hosts(
-    client: &stackable_operator::client::Client,
-    svc: &Service,
-    port_name: &str,
-) -> Result<impl IntoIterator<Item = (String, u16)>, Error> {
-    let svc_port = svc
-        .spec
-        .as_ref()
-        .and_then(|svc_spec| {
-            svc_spec
-                .ports
-                .as_ref()?
-                .iter()
-                .find(|port| port.name.as_deref() == Some(HIVE_PORT_NAME))
-        })
-        .context(NoServicePortSnafu {
-            port_name,
-            obj_ref: ObjectRef::from_obj(svc),
-        })?;
-
-    let node_port = svc_port.node_port.context(NoNodePortSnafu {
-        port_name,
-        obj_ref: ObjectRef::from_obj(svc),
-    })?;
-    let endpoints = client
-        .get::<Endpoints>(
-            svc.metadata.name.as_deref().context(NoNameSnafu)?,
-            svc.metadata
-                .namespace
-                .as_deref()
-                .context(NoNamespaceSnafu)?,
-        )
-        .await
-        .with_context(|_| FindEndpointsSnafu {
-            svc: ObjectRef::from_obj(svc),
-        })?;
-    let nodes = endpoints
-        .subsets
-        .into_iter()
-        .flatten()
-        .flat_map(|subset| subset.addresses)
-        .flatten()
-        .flat_map(|addr| addr.node_name);
-    let addrs = nodes
-        .map(|node| Ok((node, node_port.try_into().context(InvalidNodePortSnafu)?)))
-        .collect::<Result<BTreeSet<_>, _>>()?;
-    Ok(addrs)
+pub fn build_headless_listener_service_name(name: String) -> String {
```

Comment:
Just flagging this, as I felt I saw something about the name of this function earlier.

Reply:
Ah yeah: https://github.com/stackabletech/hive-operator/pull/605#discussion_r2161059271

Which is now done.